misk/src/test/kotlin/misk/client/HttpClientEnvoyTest.kt

```diff
@@ -9,42 +8,36 @@ import misk.MiskServiceModule
 import misk.inject.KAbstractModule
 import misk.testing.MiskTest
 import misk.testing.MiskTestModule
-import okhttp3.Protocol
 import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
 import org.assertj.core.api.Assertions.assertThat
-import org.junit.jupiter.api.Disabled
 import org.junit.jupiter.api.Test
 import retrofit2.Call
 import retrofit2.http.Body
 import retrofit2.http.POST
 import java.io.File
-import java.util.Arrays
 import javax.inject.Inject
 import javax.inject.Singleton
 
 /** Demonstrates ability to bind up an envoy-based http client and connect via unix sockets. */
 @MiskTest(startService = true)
 internal class HttpClientEnvoyTest {
   @MiskTestModule val module = TestModule()
-
-  @Inject private lateinit var webServerService: MockWebServerService
   @Inject @Named("dinosaur") private lateinit var client: DinosaurService
 
-  // TODO(nb): Fix MockWebServer to support this.
-  //
-  // The test as-is fails with the following exception:
+  // TODO(nb): Fix MockWebServer to support testing over Unix Socket
+  // Below test with unix sockets fails because of:
   // jnr.unixsocket.UnixSocketAddress cannot be cast to java.net.InetSocketAddress
   // java.lang.ClassCastException: jnr.unixsocket.UnixSocketAddress cannot be cast to java.net.InetSocketAddress
   //   at okhttp3.internal.http2.Http2Connection$Builder.socket(Http2Connection.java:559)
   //   at okhttp3.mockwebserver.MockWebServer$3.processConnection(MockWebServer.java:501)
-  @Disabled("MockWebServer + HTTP/2 + Unix Sockets not playing together nicely")
-  @Test fun useEnvoyClient() {
 
+  @Test fun useEnvoyClientOverHttp() {
     val dinoRequest = Dinosaur.Builder().name("dinoRequest").build()
 
-    val server = webServerService.server!!
-    server.setProtocols(Arrays.asList(Protocol.H2_PRIOR_KNOWLEDGE))
+    val server = MockWebServer()
```

Comment:
Per the OKHttp docs:

>   // Create a MockWebServer. These are lean enough that you can create a new
>   // instance for every unit test.

https://github.com/square/okhttp/tree/master/mockwebserver

Has the added benefit of making it easier to manipulate the server before starting.